plugins/module_utils/azure_rm_common.py

```diff
@@ -874,7 +875,10 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None, supp
         self.log('Getting management service client {0}'.format(client_type.__name__))
         self.check_client_version(client_type)
 
-        client_argspec = inspect.getargspec(client_type.__init__)
+        if sys.version.startswith('3.11'):
```

Comment:
This condition check should be less specific to match true on versions of python greater than or equal to 3.11.   Maybe switch to sys.version_info and check the individual components?